processing/src/main/java/org/apache/druid/segment/incremental/IncrementalIndex.java

```diff
@@ -557,12 +573,17 @@ IncrementalIndexRowResult toIncrementalIndexRow(InputRow row)
         DimensionIndexer indexer = desc.getIndexer();
         Object dimsKey = null;
         try {
-          dimsKey = indexer.processRowValsToUnsortedEncodedKeyComponent(row.getRaw(dimension), true);
+          final EncodedDimensionValue<?> encodedDimensionValue
+              = indexer.processRowValsToUnsortedEncodedKeyComponent(row.getRaw(dimension), true);
```

Comment:
i guess due to the way this refactor was done (compared to aggs) there is no real way to turn off calculating the estimates, even if we aren't using them. maybe it doesn't matter if there is no/minimal performance impact

Reply:
Yes, I was not too sure about this either. I will take another and see if we can separate the two flows without too much duplication.

Reply:
Updated `DimensionDictionary` and `StringDimensionIndexer` to turn off estimations if not needed.